packages/flutter_tools/test/integration.shard/test_driver.dart

```diff
@@ -423,14 +423,11 @@ abstract class FlutterTestDriver {
         final StringBuffer error = StringBuffer();
         error.write('Received app.stop event while waiting for $interestingOccurrence\n\n$_errorBuffer');
         final Object? jsonParams = json['params'];
-        if (jsonParams is Map<String, Object?>) {
-          if (jsonParams['error'] != null) {
-            error.write('${jsonParams['error']}\n\n');
-          }
-          final Object? trace = jsonParams['trace'];
-          if (trace != null) {
-            error.write('$trace\n\n');
-          }
+        if (jsonParams case {'error': final Object errorObject}) {
+          error.write('$errorObject\n\n');
+        }
```

Comment:
This is another usage of patterns I think I like, but I wonder if this contradicts some of my earlier thoughts. I think I'm okay with this change since the code is interested in both what `jsonParams` is as well as what `jsonParams['error`]` is.